content/blog/aws-rds-blue-green-deployment-updates/index.md

```diff
@@ -0,0 +1,91 @@
+
+## How it works
+
+By default, updates to a database will be applied in-place, meaning there may be service interruptions while a database is modified especially when making long-running updates like changing the engine version, parameter groups, or storage settings.
+
+`blueGreenUpdate` can be enabled for [Low-Downtime Updates](https://www.pulumi.com/registry/packages/aws/api-docs/rds/instance/#low-downtime-updates). When this setting is on and an update is performed, the program starts the update by creating a blue/green deployment to maintain availability. Then, it performs the requested update on the green instance, performs a guardrail check, promotes the green instance, and finally deletes the old instance and removes the blue/green deployment. This leaves just the updated database running with minimal downtime during the update while minimizing the downtime.
+
+Though this strategy typically increases the amount of time that updates need to run, it minimizes downtime for the application by keeping a current version of the database available during the update process.
+
+## How to use
+
+Please see the full example code here: [Blue/Green Deployment Updates on RDS](https://github.com/pulumi-demos/examples/tree/main/typescript/aws-rds-blue-green-updates). To use blue/green deployments for your database updates, start by creating a database and corresponding parameter group.
+
+```typescript
+const parameterGroup = new aws.rds.ParameterGroup(
+  `${baseName}logicalreplicationgroup`,
+  {
+    family: `postgres${dbVersion.split(".")[0]}`,
+    parameters: [
+      {
+        name: "rds.logical_replication",
+        value: "1",
+        applyMethod: "pending-reboot",
+      },
+    ],
+  },
+  { deleteBeforeReplace: false }
+);
+
+const database = new aws.rds.Instance(
+  baseName,
+  {
+    allocatedStorage: storageAllocation,
+    engine: "postgres",
+    engineVersion: dbVersion,
+    instanceClass: aws.rds.InstanceType.T3_Micro,
+    username: dbUsername,
+    password: dbPW,
+    storageEncrypted: true,
+    backupRetentionPeriod: 7,
+    blueGreenUpdate: { enabled: true },
+    skipFinalSnapshot: true,
+    parameterGroupName: parameterGroup.name,
+  },
+  { dependsOn: [parameterGroup] }
+);
+```
+
+After running `pulumi up` with the above code, you'll have a database deployed and attached to your custom parameter group. It's important to create and attach a custom parameter group so that you can keep it up to date with any engine version upgrades. If you use the default parameter group, updates to the database version may fail when the green deployment is created since it's parameter group family won't match the new version. You can avoid a failure by ensuring that the parameter group's `family` and `engineVersion` are always changed in the same update.
```

Comment:
```suggestion
After running `pulumi up` with the above code, you'll have a database deployed and attached to your custom parameter group. It's important to create and attach a custom parameter group so that you can keep it up to date with any engine version upgrades. If you use the default parameter group, updates to the database version may fail when the green deployment is created since its parameter group family won't match the new version. You can avoid a failure by ensuring that the parameter group's `family` and `engineVersion` are always changed in the same update.
```